crates/extrinsics/src/call.rs

```diff
@@ -16,278 +16,364 @@
+    pub fn message<T: Into<String>>(
+        self,
+        message: T,
+    ) -> CallCommandBuilder<state::Message, E> {
+        CallCommandBuilder {
+            opts: CallOpts {
+                message: message.into(),
+                ..self.opts
+            },
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<M> CallCommandBuilder<M, Missing<state::ExtrinsicOptions>> {
+    /// Sets the extrinsic operation.
+    pub fn extrinsic_opts(
+        self,
+        extrinsic_opts: ExtrinsicOpts,
+    ) -> CallCommandBuilder<M, state::ExtrinsicOptions> {
+        CallCommandBuilder {
+            opts: CallOpts {
+                extrinsic_opts,
+                ..self.opts
+            },
+            marker: PhantomData,
+        }
     }
+}
 
-    pub fn run(&self) -> Result<(), ErrorVariant> {
-        let artifacts = self.extrinsic_opts.contract_artifacts()?;
-        let transcoder = artifacts.contract_transcoder()?;
+impl<M, E> CallCommandBuilder<M, E> {
+    /// Sets the the address of the the contract to call.
+    pub fn contract(self, contract: <DefaultConfig as Config>::AccountId) -> Self {
+        let mut this = self;
+        this.opts.contract = contract;
+        this
+    }
 
-        let call_data = transcoder.encode(&self.message, &self.args)?;
+    /// Sets the arguments of the contract message to call.
+    pub fn args<T: ToString>(self, args: Vec<T>) -> Self {
+        let mut this = self;
+        this.opts.args = args.into_iter().map(|arg| arg.to_string()).collect();
+        this
+    }
+
+    /// Sets the maximum amount of gas to be used for this command.
+    pub fn gas_limit(self, gas_limit: Option<u64>) -> Self {
+        let mut this = self;
+        this.opts.gas_limit = gas_limit;
+        this
+    }
+
+    /// Sets the maximum proof size for this call.
+    pub fn proof_size(self, proof_size: Option<u64>) -> Self {
+        let mut this = self;
+        this.opts.proof_size = proof_size;
+        this
+    }
+
+    /// Sets the value to be transferred as part of the call.
+    pub fn value(self, value: BalanceVariant) -> Self {
+        let mut this = self;
+        this.opts.value = value;
+        this
+    }
+}
+
+impl CallCommandBuilder<state::Message, state::ExtrinsicOptions> {
+    /// Preprocesses contract artifacts and options for subsequent contract calls.
+    ///
+    /// This function prepares the necessary data for making a contract call based on the
+    /// provided contract artifacts, message, arguments, and options. It ensures that the
+    /// required contract code and message data are available, sets up the client, signer,
+    /// and other relevant parameters, preparing for the contract call operation.
+    ///
+    /// Returns the `CallExec` containing the preprocessed data for the contract call,
+    /// or an error in case of failure.
+    pub async fn done(self) -> CallExec {
+        let artifacts = self.opts.extrinsic_opts.contract_artifacts().unwrap();
```

Comment:
This and other `done` methods should return `Result` instead of having all these `unwrap`s